### ISSUE TITLE
Parse failures

### ISSUE THREAD

**Issue by cocreature** [GaloisInc/llvm-pretty-bc-parser#4]
Hey, thanks for the package. I got two parse erros when trying to analyze some bitcode generated by compiling the gnu coreutils. I used llvm/clang 3.6.2 to generate the assembly and the bitcode.
The first one is from `printf-frexpl.c` with the [human readable assembly](http://lpaste.net/140342) and the [bitcode](http://purelyfunctional.org/downloads/printf-frexpl.bc).
and here's the error message

```
Error {errContext = ["CONSTANTS_BLOCK","printf_frexpl","FUNCTION_BLOCK","MODULE_BLOCK"], errMessage = "unknown constant record code: 23"}
```

The second error is in `parse-datetime.c` with the [human readable assembly](http://lpaste.net/140343) and the [bitcode](http://purelyfunctional.org/downloads/parse-datetime.bc).
The error message is fairly long

```
Error {errContext = ["getFnValueById","getValue","FUNC_CODE_INST_SWITCH","yyparse","FUNCTION_BLOCK","MODULE_BLOCK"], errMessage = "entry 1302 is missing from the symbol table\nfromList [(SymTabEntry 0,Left \"yyparse.yyval_default\"),(SymTabEntry 1,Left \"yypact\"),(SymTabEntry 2,Left \"yytranslate\"),(SymTabEntry 3,Left \"yycheck\"),(SymTabEntry 4,Left \"yytable\"),(SymTabEntry 5,Left \"yydefact\"),(SymTabEntry 6,Left \"yyr2\"),(SymTabEntry 7,Left \"yyr1\"),(SymTabEntry 8,Left \"yypgoto\"),(SymTabEntry 9,Left \"yydefgoto\"),(SymTabEntry 10,Left \".str6\"),(SymTabEntry 11,Left \".str7\"),(SymTabEntry 12,Left \".str8\"),(SymTabEntry 13,Left \".str9\"),(SymTabEntry 14,Left \".str10\"),(SymTabEntry 15,Left \"meridian_table\"),(SymTabEntry 16,Left \"month_and_day_table\"),(SymTabEntry 17,Left \"dst_table\"),(SymTabEntry 18,Left \"time_units_table\"),(SymTabEntry 19,Left \"relative_time_table\"),(SymTabEntry 20,Left \"military_table\"),(SymTabEntry 21,Left \".str11\"),(SymTabEntry 22,Left \".str12\"),(SymTabEntry 23,Left \".str13\"),(SymTabEntry 24,Left \".str14\"),(SymTabEntry 25,Left \".str15\"),(SymTabEntry 26,Left \".str16\"),(SymTabEntry 27,Left \".str17\"),(SymTabEntry 28,Left \".str18\"),(SymTabEntry 29,Left \".str19\"),(SymTabEntry 30,Left \".str20\"),(SymTabEntry 31,Left \".str21\"),(SymTabEntry 32,Left \".str22\"),(SymTabEntry 33,Left \".str23\"),(SymTabEntry 34,Left \".str24\"),(SymTabEntry 35,Left \".str25\"),(SymTabEntry 36,Left \".str26\"),(SymTabEntry 37,Left \".str27\"),(SymTabEntry 38,Left \".str28\"),(SymTabEntry 39,Left \".str29\"),(SymTabEntry 40,Left \".str30\"),(SymTabEntry 41,Left \".str31\"),(SymTabEntry 42,Left \".str32\"),(SymTabEntry 43,Left \".str33\"),(SymTabEntry 44,Left \".str34\"),(SymTabEntry 45,Left \".str35\"),(SymTabEntry 46,Left \".str36\"),(SymTabEntry 47,Left \".str37\"),(SymTabEntry 48,Left \".str38\"),(SymTabEntry 49,Left \".str39\"),(SymTabEntry 50,Left \".str40\"),(SymTabEntry 51,Left \".str41\"),(SymTabEntry 52,Left \".str42\"),(SymTabEntry 53,Left \".str43\"),(SymTabEntry 54,Left \".str44\"),(SymTabEntry 55,Left \".str45\"),(SymTabEntry 56,Left \".str46\"),(SymTabEntry 57,Left \".str47\"),(SymTabEntry 58,Left \".str48\"),(SymTabEntry 59,Left \".str49\"),(SymTabEntry 60,Left \".str50\"),(SymTabEntry 61,Left \".str51\"),(SymTabEntry 62,Left \".str52\"),(SymTabEntry 63,Left \".str53\"),(SymTabEntry 64,Left \".str54\"),(SymTabEntry 65,Left \".str55\"),(SymTabEntry 66,Left \".str56\"),(SymTabEntry 67,Left \".str57\"),(SymTabEntry 68,Left \".str58\"),(SymTabEntry 69,Left \".str59\"),(SymTabEntry 70,Left \".str60\"),(SymTabEntry 71,Left \".str61\"),(SymTabEntry 72,Left \".str62\"),(SymTabEntry 73,Left \".str63\"),(SymTabEntry 74,Left \".str64\"),(SymTabEntry 75,Left \".str65\"),(SymTabEntry 76,Left \".str66\"),(SymTabEntry 77,Left \"universal_time_zone_table\"),(SymTabEntry 78,Left \"time_zone_table\"),(SymTabEntry 79,Left \".str67\"),(SymTabEntry 80,Left \".str68\"),(SymTabEntry 81,Left \".str69\"),(SymTabEntry 82,Left \".str70\"),(SymTabEntry 83,Left \".str71\"),(SymTabEntry 84,Left \".str72\"),(SymTabEntry 85,Left \".str73\"),(SymTabEntry 86,Left \".str74\"),(SymTabEntry 87,Left \".str75\"),(SymTabEntry 88,Left \".str76\"),(SymTabEntry 89,Left \".str77\"),(SymTabEntry 90,Left \".str78\"),(SymTabEntry 91,Left \".str79\"),(SymTabEntry 92,Left \".str80\"),(SymTabEntry 93,Left \".str81\"),(SymTabEntry 94,Left \".str82\"),(SymTabEntry 95,Left \".str83\"),(SymTabEntry 96,Left \".str84\"),(SymTabEntry 97,Left \".str85\"),(SymTabEntry 98,Left \".str86\"),(SymTabEntry 99,Left \".str87\"),(SymTabEntry 100,Left \".str88\"),(SymTabEntry 101,Left \".str89\"),(SymTabEntry 102,Left \".str90\"),(SymTabEntry 103,Left \".str91\"),(SymTabEntry 104,Left \".str92\"),(SymTabEntry 105,Left \".str93\"),(SymTabEntry 106,Left \".str94\"),(SymTabEntry 107,Left \".str95\"),(SymTabEntry 108,Left \".str96\"),(SymTabEntry 109,Left \".str97\"),(SymTabEntry 110,Left \".str98\"),(SymTabEntry 111,Left \".str99\"),(SymTabEntry 112,Left \".str100\"),(SymTabEntry 113,Left \".str101\"),(SymTabEntry 114,Left \".str102\"),(SymTabEntry 115,Left \".str103\"),(SymTabEntry 116,Left \".str104\"),(SymTabEntry 117,Left \".str105\"),(SymTabEntry 118,Left \".str106\"),(SymTabEntry 119,Left \".str107\"),(SymTabEntry 120,Left \".str108\"),(SymTabEntry 121,Left \".str109\"),(SymTabEntry 122,Left \".str110\"),(SymTabEntry 123,Left \".str111\"),(SymTabEntry 124,Left \".str112\"),(SymTabEntry 125,Left \".str113\"),(SymTabEntry 126,Left \".str114\"),(SymTabEntry 127,Left \".str115\"),(SymTabEntry 128,Left \".str116\"),(SymTabEntry 129,Left \".str117\"),(SymTabEntry 130,Left \".str118\"),(SymTabEntry 131,Left \".str119\"),(SymTabEntry 132,Left \".str120\"),(SymTabEntry 133,Left \".str121\"),(SymTabEntry 134,Left \".str122\"),(SymTabEntry 135,Left \".str123\"),(SymTabEntry 136,Left \".str124\"),(SymTabEntry 137,Left \".str125\"),(SymTabEntry 138,Left \".str126\"),(SymTabEntry 139,Left \".str127\"),(SymTabEntry 140,Left \".str128\"),(SymTabEntry 141,Left \".str129\"),(SymTabEntry 142,Left \".str130\"),(SymTabEntry 143,Left \".str131\"),(SymTabEntry 144,Left \".str132\"),(SymTabEntry 145,Left \".str133\"),(SymTabEntry 146,Left \".str134\"),(SymTabEntry 147,Left \".str135\"),(SymTabEntry 148,Left \".str136\"),(SymTabEntry 149,Left \".str137\"),(SymTabEntry 150,Left \".str138\"),(SymTabEntry 151,Left \".str139\"),(SymTabEntry 152,Left \".str140\"),(SymTabEntry 153,Left \".str141\"),(SymTabEntry 154,Left \".str142\"),(SymTabEntry 155,Left \".str143\"),(SymTabEntry 156,Left \".str144\"),(SymTabEntry 157,Left \"yyparse\"),(SymTabEntry 158,Left \"llvm.lifetime.start\"),(SymTabEntry 159,Left \"llvm.memcpy.p0i8.p0i8.i64\"),(SymTabEntry 160,Left \"free\"),(SymTabEntry 161,Left \"llvm.lifetime.end\"),(SymTabEntry 162,Left \"parse_datetime\"),(SymTabEntry 163,Left \"gettime\"),(SymTabEntry 164,Left \"localtime\"),(SymTabEntry 165,Left \"strncmp\"),(SymTabEntry 166,Left \"xmalloc\"),(SymTabEntry 167,Left \"setenv\"),(SymTabEntry 168,Left \"strcmp\"),(SymTabEntry 169,Left \"mktime\"),(SymTabEntry 170,Left \"__sprintf_chk\"),(SymTabEntry 171,Left \"unsetenv\"),(SymTabEntry 172,Left \"getenv\"),(SymTabEntry 173,Left \"strlen\"),(SymTabEntry 174,Left \"xmemdup\"),(SymTabEntry 175,Left \"lookup_zone\"),(SymTabEntry 176,Left \"llvm.uadd.with.overflow.i64\"),(SymTabEntry 177,Left \"llvm.memset.p0i8.i64\"),(SymTabEntry 697,Left \"pc\"),(SymTabEntry 960,Left \"buff.i\"),(SymTabEntry 961,Left \"yyssa\"),(SymTabEntry 962,Left \"yyvsa\"),(SymTabEntry 1010,Left \"yyvsp.0\"),(SymTabEntry 1011,Left \"yyssp.0\"),(SymTabEntry 1012,Left \"yystate.0\"),(SymTabEntry 1013,Left \"yychar.0\"),(SymTabEntry 1017,Left \"yychar.1295\"),(SymTabEntry 1018,Left \"yystate.1294\"),(SymTabEntry 1019,Left \"yyssp.1290\"),(SymTabEntry 1020,Left \"yyvsp.1288\"),(SymTabEntry 1026,Left \"yyresult.0259.ph\"),(SymTabEntry 1033,Left \".pre.i\"),(SymTabEntry 1034,Left \".lcssa351\"),(SymTabEntry 1041,Left \".be\"),(SymTabEntry 1046,Left \".lcssa352\"),(SymTabEntry 1047,Left \".lcssa\"),(SymTabEntry 1053,Left \"sign.0.i\"),(SymTabEntry 1054,Left \"c.0.i\"),(SymTabEntry 1058,Left \"p.062.i\"),(SymTabEntry 1068,Left \"uadd.i\"),(SymTabEntry 1071,Left \".lcssa367\"),(SymTabEntry 1072,Left \".lcssa365\"),(SymTabEntry 1073,Left \"p.062.i.lcssa\"),(SymTabEntry 1074,Left \".lcssa362\"),(SymTabEntry 1084,Left \"s.0.i\"),(SymTabEntry 1093,Left \".pre95.i\"),(SymTabEntry 1095,Left \"ns.1.i\"),(SymTabEntry 1097,Left \"p.2.i\"),(SymTabEntry 1106,Left \".lcssa370\"),(SymTabEntry 1107,Left \"ns.2.ph.i\"),(SymTabEntry 1108,Left \"p.4.ph.i\"),(SymTabEntry 1110,Left \"p.357.i\"),(SymTabEntry 1112,Left \"p.357.i.lcssa\"),(SymTabEntry 1119,Left \"p.4.i\"),(SymTabEntry 1125,Left \"p.4.i.lcssa\"),(SymTabEntry 1127,Left \"or.cond22.i\"),(SymTabEntry 1131,Left \"s.1.i\"),(SymTabEntry 1132,Left \"ns.3.i\"),(SymTabEntry 1140,Left \"sign.0.lobit.i\"),(SymTabEntry 1152,Left \".off.i\"),(SymTabEntry 1154,Left \".lcssa357\"),(SymTabEntry 1155,Left \".lcssa353\"),(SymTabEntry 1157,Left \"p4.0.i\"),(SymTabEntry 1158,Left \"c.2.i\"),(SymTabEntry 1161,Left \".pre93.i\"),(SymTabEntry 1163,Left \"p4.1.i\"),(SymTabEntry 1168,Left \".off34.i\"),(SymTabEntry 1171,Left \"or.cond32.i\"),(SymTabEntry 1172,Left \"p4.1.i.lcssa\"),(SymTabEntry 1178,Left \"p.036.i.i\"),(SymTabEntry 1179,Left \".off.i.i\"),(SymTabEntry 1182,Left \"..i.i\"),(SymTabEntry 1193,Left \"tp.134.i.i\"),(SymTabEntry 1222,Left \"p.2.i68.i\"),(SymTabEntry 1223,Left \"period_found.1.i67.i\"),(SymTabEntry 1224,Left \".pre48.i.i\"),(SymTabEntry 1227,Left \"period_found.1.i.i\"),(SymTabEntry 1228,Left \"p.2.i.i\"),(SymTabEntry 1232,Left \"period_found.1.i.i.lcssa\"),(SymTabEntry 1233,Left \"period_found.1.i.lcssa.i\"),(SymTabEntry 1234,Left \"phitmp.i.i\"),(SymTabEntry 1235,Left \"phitmp39.i.i\"),(SymTabEntry 1261,Left \"period_found.1.i65.i\"),(SymTabEntry 1262,Left \"p.2.i66.i\"),(SymTabEntry 1347,Left \"tp.134.i.i.lcssa\"),(SymTabEntry 1348,Left \".0.i.ph.i\"),(SymTabEntry 1358,Left \".lcssa354\"),(SymTabEntry 1363,Left \"count.0.i\"),(SymTabEntry 1368,Left \"count.1.i\"),(SymTabEntry 1372,Left \".pre96.i\"),(SymTabEntry 1374,Left \"ns.1.1.i\"),(SymTabEntry 1375,Left \"p.2.1.i\"),(SymTabEntry 1382,Left \".pre97.i\"),(SymTabEntry 1384,Left \"ns.1.2.i\"),(SymTabEntry 1385,Left \"p.2.2.i\"),(SymTabEntry 1392,Left \".pre98.i\"),(SymTabEntry 1394,Left \"ns.1.3.i\"),(SymTabEntry 1395,Left \"p.2.3.i\"),(SymTabEntry 1402,Left \".pre99.i\"),(SymTabEntry 1404,Left \"ns.1.4.i\"),(SymTabEntry 1405,Left \"p.2.4.i\"),(SymTabEntry 1412,Left \".pre100.i\"),(SymTabEntry 1414,Left \"ns.1.5.i\"),(SymTabEntry 1415,Left \"p.2.5.i\"),(SymTabEntry 1422,Left \".pre101.i\"),(SymTabEntry 1424,Left \"ns.1.6.i\"),(SymTabEntry 1425,Left \"p.2.6.i\"),(SymTabEntry 1432,Left \"ns.1.7.i\"),(SymTabEntry 1433,Left \"p.2.7.i\"),(SymTabEntry 1438,Left \".0.i\"),(SymTabEntry 1443,Left \"yychar.2\"),(SymTabEntry 1450,Left \"yytoken.0\"),(SymTabEntry 1451,Left \"yychar.3\"),(SymTabEntry 1479,Left \"yychar.4\"),(SymTabEntry 1488,Left \"yyn.0\"),(SymTabEntry 1489,Left \"yychar.5\"),(SymTabEntry 1578,Left \"or.cond.i\"),(SymTabEntry 1590,Left \"n_minutes.0.i\"),(SymTabEntry 1592,Left \"ispos.i\"),(SymTabEntry 1593,Left \"neg.i\"),(SymTabEntry 1675,Left \"or.cond.i196\"),(SymTabEntry 1687,Left \"n_minutes.0.i198\"),(SymTabEntry 1689,Left \"ispos.i199\"),(SymTabEntry 1690,Left \"neg.i200\"),(SymTabEntry 1797,Left \"sext\"),(SymTabEntry 1942,Left \"or.cond\"),(SymTabEntry 1959,Left \".sink1.i\"),(SymTabEntry 1960,Left \".sink.i\"),(SymTabEntry 1975,Left \"or.cond261\"),(SymTabEntry 1992,Left \".sink1.i202\"),(SymTabEntry 1993,Left \".sink.i203\"),(SymTabEntry 2020,Left \"yyval.sroa.0.0\"),(SymTabEntry 2021,Left \"yyval.sroa.39.0\"),(SymTabEntry 2022,Left \"yyval.sroa.72.0\"),(SymTabEntry 2023,Left \"yyval.sroa.104.0\"),(SymTabEntry 2024,Left \"yyval.sroa.135.0\"),(SymTabEntry 2025,Left \"yyval.sroa.166.0\"),(SymTabEntry 2026,Left \"yyval.sroa.199.0\"),(SymTabBBEntry 0,Left \".lr.ph\"),(SymTabBBEntry 3,Left \".thread257.preheader\"),(SymTabBBEntry 7,Left \".backedge.i.loopexit\"),(SymTabBBEntry 8,Left \".backedge.i\"),(SymTabBBEntry 9,Left \".critedge25.i\"),(SymTabBBEntry 10,Left \"switch.early.test.i\"),(SymTabBBEntry 11,Left \"switch.edge.i\"),(SymTabBBEntry 12,Left \".backedge.i.backedge\"),(SymTabBBEntry 14,Left \".critedge29.i.preheader\"),(SymTabBBEntry 15,Left \".critedge29.i\"),(SymTabBBEntry 16,Left \".critedge29.i.backedge\"),(SymTabBBEntry 17,Left \".critedge30.i\"),(SymTabBBEntry 18,Left \".lr.ph63.i\"),(SymTabBBEntry 30,Left \".preheader35.i\"),(SymTabBBEntry 31,Left \".lr.ph.i.preheader\"),(SymTabBBEntry 32,Left \".preheader.i.loopexit\"),(SymTabBBEntry 33,Left \".preheader.i\"),(SymTabBBEntry 34,Left \".lr.ph.i\"),(SymTabBBEntry 47,Left \".critedge.i.preheader\"),(SymTabBBEntry 48,Left \".critedge.i\"),(SymTabBBEntry 52,Left \".lr.ph38.i.i.preheader\"),(SymTabBBEntry 53,Left \".preheader19.i.i.loopexit\"),(SymTabBBEntry 54,Left \".preheader19.i.i\"),(SymTabBBEntry 55,Left \".lr.ph38.i.i\"),(SymTabBBEntry 67,Left \".preheader16.i.i\"),(SymTabBBEntry 71,Left \".preheader13.i.i\"),(SymTabBBEntry 73,Left \".preheader12.i.i\"),(SymTabBBEntry 74,Left \".preheader.i.i\"),(SymTabBBEntry 76,Left \"._crit_edge47.i.i\"),(SymTabBBEntry 77,Left \".lr.ph.i._crit_edge.i.loopexit\"),(SymTabBBEntry 78,Left \".lr.ph.i._crit_edge.i\"),(SymTabBBEntry 103,Left \".lr.ph.i.preheader.i\"),(SymTabBBEntry 104,Left \"._crit_edge47.i.i.preheader\"),(SymTabBBEntry 145,Left \".loopexit.i.loopexit\"),(SymTabBBEntry 146,Left \".loopexit.i\"),(SymTabBBEntry 148,Left \".preheader36.i.preheader\"),(SymTabBBEntry 150,Left \".preheader36.i\"),(SymTabBBEntry 168,Left \"yylex.exit.loopexit\"),(SymTabBBEntry 169,Left \"yylex.exit.loopexit350\"),(SymTabBBEntry 170,Left \"yylex.exit\"),(SymTabBBEntry 196,Left \"._crit_edge.i\"),(SymTabBBEntry 201,Left \"time_zone_hhmm.exit\"),(SymTabBBEntry 210,Left \"._crit_edge.i197\"),(SymTabBBEntry 215,Left \"time_zone_hhmm.exit201\"),(SymTabBBEntry 269,Left \".thread254\"),(SymTabBBEntry 280,Left \".thread256\"),(SymTabBBEntry 287,Left \"digits_to_date_time.exit204\"),(SymTabBBEntry 290,Left \"digits_to_date_time.exit\")]"}
```

**Comment by elliottt** [comment]
Thanks for the bug report, and the small example! I'll take a look at this as soon as I can, and add the example to the test suite.

**Comment by elliottt** [comment]
I've fixed the first case in 7bd63fd, and will look into the second today. Thanks again!

**Comment by atomb** [comment]
The first example is fixed, and the links to the second are dangling, so I'm going to close this.
